src/test/java/com/jayway/asyncservlet/ApplicationTests.java

```diff
@@ -30,6 +39,22 @@ public void shouldPass_springBootRepo() throws Exception {
         log.info(result.toString());
 
         assertTrue(op.getStatusCode().is2xxSuccessful());
-
+    }
+    
+    @Import(Application.class)
+    @Configuration
+    public static class TestConfig {
+        @Bean
+        RepoListService<GitHubItems, RepoListDto> buildRepoListService(AsyncRestTemplate asyncRestTemplate){
+            String json = "{ \"total_count\": 2, \"items\": ["
```

Comment:
I like your thinking - this decouples the data from java pojos.
I will refactor this after the PR to show a slightly better way of doing the json part (imho).